### ISSUE TITLE
Difficulty reproducing ULD (TOFU) paper metrics

### ISSUE THREAD

**Issue by QinKaiii** [UCSB-NLP-Chang/ULD#2]
Hi authors,

First of all, thank you for sharing your excellent work on "Reversing the Forget-Retain Objectives: An Efficient LLM Unlearning Framework from Logit Difference" (ULD) and for making the code publicly available! We have been running the code provided in this repository, carefully following the installation and execution instructions in the `README.md` for the TOFU dataset.

According to Table [2] in your paper, the expected performance for the **ULD method** on the TOFU dataset is as follows:

*   **TOFU-1% forget (ULD):**
    *   FQ: **0.99**
    *   MU: **0.62**
*   **TOFU-5% forget (ULD):**
    *   FQ: **0.73**
    *   MU: **0.62**

However, when we run the ULD experiments, the results we are obtaining are significantly different, especially for Forget Quality:

*   **Our results for TOFU-1% forget (ULD):**
    *   FQ: **0.003**
    *   MU: **0.65**
*   **Our results for TOFU-5% forget (ULD):**
    *   FQ: **1.33e-13**
    *   MU: **0.648**

Interestingly, when we run the **NPO method** (also implemented in this repository) using a similar setup based on your scripts, the performance metrics we obtain for NPO align well with those reported in your paper for NPO.

Given this, we were wondering **if there might be a specific branch, commit, or a particular set of hyperparameters/configurations that are crucial for replicating the reported ULD results, which might differ from the NPO setup or the general README instructions for ULD**?

We would be very grateful if you could provide some guidance or clarification specifically for reproducing the ULD results on TOFU. We are happy to provide any further information if needed.

Thank you for your time and assistance!

Best regards

**Comment by Question406** [comment]
Hi, 

From our experience, FQ is a very sensitive metric as it is the p-value of two distributions. We observed a significant difference when torch/transformer version are different under same training hyper-parameters, even under different GPUs, for example, on A100 and A6000. To improve it, we believe a sweep over different hyper-parameter like lr, batch size, can be an effective way. And seems some recent unlearning papers on TOFU already employs other metrics for evaluating unlearn performance.

Hope this response helps you!
